src/util/support/utf8_conv.c

```diff
@@ -99,24 +98,16 @@ k5_utf8s_to_ucs2s(krb5_ucs2 *ucs2str,
             if ((utf8str[i] & 0xc0) != 0x80)
                 return -1;
 
-            ch <<= 6;                   /* 6 bits of data in each subsequent byte */
+            ch <<= 6; /* 6 bits of data in each subsequent byte */
             ch |= (krb5_ucs2)(utf8str[i] & 0x3f);
         }
 
         if (ucs2str != NULL) {
-#ifdef K5_BE
-#ifndef SWAP16
-#define SWAP16(X)       ((((X) << 8) | ((X) >> 8)) & 0xFFFF)
-#endif
-            if (little_endian)
-                ucs2str[ucs2len] = SWAP16(ch);
-            else
-#endif
-                ucs2str[ucs2len] = ch;
+            ucs2str[ucs2len] = load_16_le(&ch);
```

Comment:
Something is wrong here (and in the old code), although I will have to look more closely to see how it ought to work.  load_16_le() must only be applied to a pointer to two bytes, not to a 16-bit value.  We should not be constructing a krb5_ucs2 array filled with byte-swaped values; we should produce a byte array and use store_16_le() to write the 16-bit values into it.